### PR TITLE
Support for CircleCI

### LINKED ISSUE #1157: Set up CircleCI for running workflows

### PR DESCRIPTION
**Pull request by chrisfilo**
This PR adds support for extensive continuous integration testing that involves:

* building docs and checking for syntax errors
* running unit tests comparing FSL and our pipelines on FSL's course data
* running four FSL/SPM pipelines on our example data and FEEDS

The tests are run for free on CircleCI. They will run for every PR, but it takes around 4h for everything to compute. The results are integrated with GitHub the same way as Travis.

The PR also includes many fixes I had to do to make those tests pass. I hope this will greatly improve the quality of the code we deliver!

Closes #1157

### CODE REVIEW

**Comment by chrisfilo** [comment]
This is ready for review and merge.

File: nipype/workflows/dmri/fsl/tests/test_dti.py [hunk 1]
```diff
@@ -11,72 +10,74 @@
...
+                                       z_size=z_size), name="slice_dwi")
+    slice_dwi.inputs.in_file = dwi_file
 
     nipype_bedpostx = create_bedpostx_pipeline("nipype_bedpostx")
-    nipype_bedpostx.inputs.inputnode.dwi = dwi_file
-    nipype_bedpostx.inputs.inputnode.mask = mask_file
     nipype_bedpostx.inputs.inputnode.bvecs = bvecs_file
     nipype_bedpostx.inputs.inputnode.bvals = bvals_file
-    nipype_bedpostx.inputs.xfibres.n_fibres = 2
+    nipype_bedpostx.inputs.xfibres.n_fibres = 1
```
Comment (satra) [inline nipype/workflows/dmri/fsl/tests/test_dti.py]: why did `n_fibres` get changed to 1?

Comment (satra) [inline nipype/workflows/dmri/fsl/tests/test_dti.py]: nevermind i see this is just for testing.

Comment (chrisfilo) [inline nipype/workflows/dmri/fsl/tests/test_dti.py]: To speed up the test.

**Pull request merged**